extensions/s3/src/main/java/io/deephaven/extensions/s3/S3InstructionsProviderPlugin.java

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.extensions.s3;
+
+import com.google.auto.service.AutoService;
+import io.deephaven.util.channel.DataInstructionsProviderPlugin;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.URI;
+import java.util.Map;
+
+/**
+ * {@link DataInstructionsProviderPlugin} implementation used for reading files from S3.
+ */
+@AutoService(DataInstructionsProviderPlugin.class)
+@SuppressWarnings("unused")
+public final class S3InstructionsProviderPlugin implements DataInstructionsProviderPlugin {
+    private static final String CLIENT_REGION = "client.region";
+    private static final String S3_ACCESS_KEY_ID = "s3.access-key-id";
+    private static final String S3_SECRET_ACCESS_KEY = "s3.secret-access-key";
+    private static final String S3_ENDPOINT = "s3.endpoint";
```

Comment:
This logic needs to live in iceberg-s3 IMO; otherwise, we have an implicit cross project dependency and it's not clear where these properties came from. And then, we can have hard references, `AwsClientProperties.CLIENT_REGION`, `S3FileIOProperties.ACCESS_KEY_ID`, etc.